README.md

```diff
@@ -1 +1,263 @@
+  <<"one">> => #{type => ?INT}
+}).
+
+two() -> graphql:objectType(<<"Two">>, <<"Has field named two">>, #{
+  <<"two">> => #{type => ?INT}
+}).
+
+union_default() -> ?UNION(<<"UnionDefaultResolve">>, <<"Used default type resolver">>, [
+  fun one/0,
+  fun two/0
+]).
+
+union_custom() -> ?UNION(<<"UnionCastomResolve">>, <<"Used custom type resolver">>, [
+  fun one/0,
+  fun two/0
+], fun(Value, _ThisTypeDefinition)->
+  Type = case maps:get(<<"one">>, Value, undefined) of
+    undefined -> fun two/0;
+    _ -> fun one/0
+  end,
+  {Type, Value}  % resolver must return tuple {ResolvedType, UnwrappedValue}
+end).
+```
+
+Object Type definition:
+```erlang
+test_unions() -> graphql:objectType(<<"TestUnions">>, <<>>, #{
+  <<"default_resolver">> => #{
+    type => ?LIST(fun unions_default/0),
+    
+    % default resolver just compare first elem in tuple with possibleTypes in union  
+    resolver => fun() ->
+      [
+        {fun one/0, #{<<"one">> => 1}},
+        {fun two/0, #{<<"two">> => 2}},
+        {fun one/0, #{<<"one">> => 3}},
+      ]
+    end
+  },
+  
+  <<"custom_resolver">> => #{
+    type => ?LIST(fun unions_custom/0),
+    resolver => fun()->
+      [
+        #{<<"one">> => 1},
+        #{<<"two">> => 2},
+        #{<<"one">> => 3}
+      ]
+    end
+  }
+}).
+```
+
+## Object types
+
+The most basic components of a GraphQL schema are object types, which just represent a kind
+of object you can fetch from your service, and what fields it has.
+
+Creating object type example:
+```erlang
+
+graphql:objectType(<<"Name">>, <<"Description">>, #{
+  <<"field_name">> => #{
+    type => ?STRING,  % what type of resolved value
+    description => <<"Human redable description">>,  % optional
+    isDeprecated => true, % optional. Default: false
+    deprecationReason => <<"Human redable deprecation reason">>, % used when isDeprecated => true for introspection tools
+    args => #{  % map of field arguments
+      <<"arg_name">> => #{
+        type => ?INT
+        % optional: description, isDeprecated        
+      }
+    },
+    resolver => fun() -> <<"resolved string">>
+  }
+})
+```
+
+Resolver - function for resolving field value. Arity and arguments passing:
+```erlang
+case erlang:fun_info(Resolver, arity) of
+  {arity, 0} -> Resolver();
+  {arity, 1} -> Resolver(ObjectValue);
+  {arity, 2} -> Resolver(ObjectValue, ArgumentValues);
+  {arity, 3} -> Resolver(ObjectValue, ArgumentValues, Context)
+end.
+```
+
+where:
+
+- ObjectValue - result of parent resolver (or InitialValue for query root)
+- ArgumentsValues - map of arguments. When argument not provided in query - value is 'null'
+- Context - context from parent resolver.
+
+Resolver can return resolved value witch passed to scalar serialize function if type is scalar or as ObjectValue
+for each field in child object type
+ 
+Resolver can alse overwrite context for his childrens. In this case resolver may look like:
```

Comment:
Resolver can also overwrite context of his children. In this case resolver can looks like: